app/src/main/java/org/simple/clinic/user/LoggedInUserHttpInterceptor.kt

```diff
@@ -18,9 +18,10 @@ class LoggedInUserHttpInterceptor @Inject constructor() : Interceptor {
     val user = userSession.loggedInUserImmediate()
     val (accessToken) = userSession.accessToken()
 
-    return when (user?.isApprovedForSyncing() == true) {
-      true -> chain.proceed(addHeaders(originalRequest, accessToken!!, user!!))
-      false -> chain.proceed(originalRequest)
+    return if (user != null && accessToken.isNullOrBlank().not()) {
+      chain.proceed(addHeaders(originalRequest, accessToken!!, user))
```

Comment:
TODO: understand this

Reply:
The `isApprovedForSyncing()` method checks two things: whether the `loggedInStatus` is equal to `LOGGED_IN` and the `status` is equal to `APPROVED_FOR_SYNCING`. When we call the `reset_password` api, the access token must be sent so the server knows its a valid user who's resetting the PIN, but by this time we have already set the local logged in status to `RESETTING_PIN`, so the access token does not get sent.

This change ensures that the access token gets sent regardless of whether the user is approved for syncing or not.